bin/reth/src/db/snapshots/headers.rs

```diff
@@ -0,0 +1,192 @@
+                bench_kind,
+                (open_db_read_only(db_path, log_level)?, chain.clone()),
+                jar_config,
+                || {
+                    for num in row_indexes.iter() {
+                        Header::decompress(
+                            cursor
+                                .row_by_number_with_cols::<0b01, 2>(num - self.from)?
+                                .ok_or(ProviderError::HeaderNotFound((*num as u64).into()))?[0],
+                        )?;
+                        // TODO: replace with below when eventually SnapshotProvider re-uses cursor
+                        // provider.header_by_number(num as
+                        // u64)?.ok_or(ProviderError::HeaderNotFound((*num as u64).into()))?;
+                    }
+                    Ok(())
+                },
+                |provider| {
+                    for num in row_indexes.iter() {
+                        provider
+                            .header_by_number(*num as u64)?
+                            .ok_or(ProviderError::HeaderNotFound((*num as u64).into()))?;
+                    }
+                    Ok(())
+                },
+            )?;
+
+            // For random walk
+            row_indexes.shuffle(&mut rng);
+        }
+
+        // BENCHMARK QUERYING A RANDOM HEADER BY NUMBER
+        {
+            let num = row_indexes[rng.gen_range(0..row_indexes.len())];
+            bench(
+                BenchKind::RandomOne,
+                (open_db_read_only(db_path, log_level)?, chain.clone()),
+                jar_config,
+                || {
+                    Header::decompress(
+                        cursor
+                            .row_by_number_with_cols::<0b01, 2>((num - self.from) as usize)?
+                            .ok_or(ProviderError::HeaderNotFound((num as u64).into()))?[0],
+                    )?;
+                    Ok(())
+                },
+                |provider| {
+                    provider
+                        .header_by_number(num as u64)?
+                        .ok_or(ProviderError::HeaderNotFound((num as u64).into()))?;
+                    Ok(())
+                },
+            )?;
+        }
+
+        // BENCHMARK QUERYING A RANDOM HEADER BY HASH
+        {
+            let num = row_indexes[rng.gen_range(0..row_indexes.len())] as u64;
+            let header_hash =
+                ProviderFactory::new(open_db_read_only(db_path, log_level)?, chain.clone())
+                    .header_by_number(num)?
+                    .ok_or(ProviderError::HeaderNotFound(num.into()))?
+                    .hash_slow();
+
+            bench(
+                BenchKind::RandomHash,
+                (open_db_read_only(db_path, log_level)?, chain.clone()),
+                jar_config,
+                || {
+                    let header = Header::decompress(
+                        cursor
+                            .row_by_key_with_cols::<0b01, 2>(header_hash.as_slice())?
```

Comment:
how many variants of this call do we have?

I kinda like the consts here, but if we have _a lot_ of them perhaps we should consider args instead

Reply:
i'd reckon we wont have many tbh, but ill keep it in mind for the future